### ISSUE TITLE
Deploy on Kubernetes

### ISSUE THREAD

**Issue by nielsbasjes** [kasmtech/workspaces-issues#213]
Because Kasm is essentially running docker images somewhere I dare to ask: 
It would be very useful if there is support for running it in Kubernetes.

**Comment by mmcclaskey** [comment]
Indeed. We have an open ticket to add support for Kubernetes and it is a high priority.

**Comment by kowalski7cc** [comment]
Yep, maybe a Kubernetes operator that deploys a fully Kubernetes/cloud native Kasm

**Comment by chris-sanders** [comment]
I'm subscribing to this, I only run K8s in my lab no virtual machines. When there's some support for deploying on K8s please be sure to update this issue or let me know if there's already a tracking issue I should be watching.

**Comment by weidi** [comment]
Love to see this as well. There is lot of potential to it!

**Comment by sk-sharif** [comment]
Iam trying to deploy kasm on Kubernetes, Is there any update on the ticket

**Comment by schmots1** [comment]
Just wondering if there is an update on the Kubernetes front.  This would be fantastic.

**Comment by j-travis** [comment]
No, sorry

**Comment by pl4nty** [comment]
I tried building this with a dind sidecar since my dev clusters use containerd/cri-o. Unfortunately the agent tried to reload nginx config via an OCI exec call to dind, and running nginx in dind might cause volume issues.

Looking forward to seeing progress on this feature, especially if it's runtime-agnostic. Runtime access can be a security concern for certain customers

**Comment by boixu** [comment]
Am also interested in this :)

**Comment by ejaysss** [comment]
Need it too.

**Comment by m00nyONE** [comment]
This would be an awesome feature because it would allow for k8s operators to spin up kasm services as needed. All fully declarative

**Comment by richevanscybermyte** [comment]
+1 on Kasm in K8s and an operator would be slick.

**Comment by rutalreja-deloitte** [comment]
+1 up

**Comment by ryanarnold1s** [comment]
+1 Would love to see this!

**Comment by subastation** [comment]
+1 bc my friend told me

**Comment by clemenko** [comment]
+1

**Comment by Liloava** [comment]
+1!

**Comment by Mr-Philipp** [comment]
+1

We are migrating everything away from docker atm.

**Comment by haliliceylan** [comment]
+1!

**Comment by godot2d** [comment]
+1！！！

**Comment by bo0tzz** [comment]
Folks, I'm just as keen to get this as the rest of you but only commenting "+1" just serves to clutter the thread. Use the upvote button on the original post ;)

**Comment by Josh-Tracy** [comment]
Me 3!

**Comment by SallehKhalid** [comment]
+1 ;)

**Comment by teja-kasm** [comment]
We’re currently enhancing our GitHub issue tracker to improve organization and efficiency.

As part of this process, we’re closing older issues to help keep the tracker focused on current and active topics.

If this issue is still relevant or you’re still experiencing the problem, please feel free to create a new issue.

Thank you for your understanding!

**Comment by especially-relative** [comment]
Is this no longer on a roadmap?

**Comment by j-travis** [comment]
We publish a helm chart that allows you to run the main core services in kubernetes. The Agent role will need to run outside the cluster for now. You can follow the progress here: 
https://github.com/kasmtech/kasm-helm
